### DISCUSSION TITLE
Trouble with 3D simulation of simple slab

### DISCUSSION THREAD

**Discussion by Srinivas-Ganti** [NanoComp/meep#2154] [Q&A] [answered]
Hi all,

I am trying to simulate a 3D structure following code similar to that shown in [this example](https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/#visualizing-3d-structures) which I am able to run without problems.

I want to do similar and first simulate a slab of oxide (eps = 2.25) and 8900 x 8900x525 um size, and a custom light source, However, I am getting a bad_alloc error. My code is below and Im getting the following error. Any suggestion on how to set it up correctly? I want to visualise the slab and cell in paraview. This worked for the previous example but not for this case...:

```
import meep as mp
import numpy as np
import math

cell_size = mp.Vector3(8925,8925,550)


vertices = [mp.Vector3(-4450,-4450),
            mp.Vector3(-4450,4450),
            mp.Vector3(4450,4450),
            mp.Vector3(4450,-4450)]

geometry = [mp.Prism(vertices, height=5, material=mp.Medium(index=2.25))]

sim = mp.Simulation(resolution=50,
                    cell_size=cell_size,
                    geometry=geometry)

sim.run(mp.at_beginning(mp.output_epsilon),
        until=10)
```
        
 

 `-----------
Initializing structure...
time for choose_chunkdivision = 1.40667e-05 s
Working in 3D dimensions.
Computational cell is 8925 x 8925 x 550 with resolution 50
     prism, center = (0,0,2.5)
          height 5, axis (0,0,1), sidewall angle: 0 radians, 4 vertices:
          (-4450,-4450,0)
          (-4450,4450,0)
          (4450,4450,0)
          (4450,-4450,0)
          dielectric constant epsilon diagonal = (4.84,4.84,4.84)
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)`

**Comment by dalarev** [comment, marked as answer]
Could it be your model is too large?

With `resolution=50`, you're looking at `50 * (8925 + 8925 + 550)` number of pixels.

**Comment by Srinivas-Ganti** [comment]
@dalarev Seems like you are right. 

I got it working by reducing the resolution and size of the computational cell

```
import meep as mp
from meep.materials import fused_quartz
import numpy as np
import math

cell_size = mp.Vector3(73,73,528)



geometry = [mp.Block(center=mp.Vector3(0,0,0), size=mp.Vector3(70,70,525), material=fused_quartz)]

sim = mp.Simulation(resolution=5,
                    cell_size=cell_size,
                    geometry=geometry)

sim.run(mp.at_beginning(mp.output_epsilon),
        until=100)
```

**Reply by smartalecH** [reply]
Note that the actual number of pixels is

(8925 * 50) * (8925 * 50) * (55 * 50)

Which is a lot...

**Reply by dalarev** [reply]
Thanks for the correction, I knew it sounded off.

**Reply by Srinivas-Ganti** [reply]
Thanks both for your answers. Do you guys know how I could do away with the fact that the materials used in my real world experiment (quartz wafer, Au patterning) could be simulated using meep? My final aim is to perform THz TDS simulation to compare with my real world samples. The quartz thickness is 525 um and the Au pattern is 200 nm thick, there are periodic structures on the top side of the quartz and a pulse of light (could use meep custom source as the real world data array here). For an example of this type of experiment see this [paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5560824/). 

How to get around the problem that the glass is so thick?

**Reply by dalarev** [reply]
The paper plots transmission, so if you wanted to recreate those results, you may have to model the full quartz depth.

In that case, perhaps implementing [symmetries](https://meep.readthedocs.io/en/latest/Exploiting_Symmetry/) could help? With the periodic structure in that paper, symmetries could be implemented in at least 2 axes.

Another option could perhaps be to look at reflection instead of transmission. If you don't think the interactions in the quartz will play a dominant role, you could shorten its depth and terminate with absorbing boundary conditions rather than PML. Note that I do not have much experience with absorbers, so I don't know if this will be physically accurate.
